### PR TITLE
Change subteams to an array and former members

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 

This PR updates the `subteam` field for the `Member` type to `subteams` which is now an array since a member can be part of multiple subteams and also renames the `otherSubteams` field to `formerSubteams`. A `formerMembers` field has also been added to the `Team` type and adds a `Former Members` section on the _Edit Teams_ page where the former members of a team can be edited. 

#### DTI Website 
No change in functionality for the DTI website, just run locally to make sure that subteams and former subteams are properly displayed. 

#### IDOL Website 
Check that operations on teams work as expected (creating, deleting, editing teams). Also check that adding/removing a member or former member correctly updates the member's own data. 
**(Test the IDOL website locally using dev database since the prod db does not have the updated fields, also the preview most likely will not work)**

### Breaking Changes 
**This PR requires database schema changes so DO NOT merge before updating the prod db**

### CODE REVIEW

File: common-types/index.d.ts [hunk 1]
```diff
@@ -27,8 +27,8 @@ interface IdolMember {
   readonly github?: string | null;
   readonly hometown: string;
   readonly about: string;
-  readonly subteam: string;
-  readonly otherSubteams?: readonly string[] | null;
+  readonly subteams: string[];
+  readonly formerSubteams?: readonly string[] | null;
```
Comment (SamChou19815) [inline common-types/index.d.ts:31]: Make all the string array type `readonly` in this file for consistency?

**Review by SamChou19815 — COMMENTED** [review]
Changes mostly look good. Just need to fix type errors.

**Comment by henry-li-06** [comment]
@SamChou19815 I think the type errors should be all good now, mostly just had to update `all-members.json` with the new fields.

File: dti-website/src/pages/team.tsx [hunk 2]
```diff
@@ -24,13 +24,13 @@ export default function TeamPage(): JSX.Element {
...
   ): readonly { info: NovaMember; id: string }[] =>
-    members
+    (members as NovaMember[])
       .filter(
         (member) =>
           ((typeof member.roleId === 'string' &&
             member.roleId.endsWith(role)) ||
             role === '') &&
-          // @ts-expect-error: missing
+          // // @ts-expect-error: missing
```
Comment (SamChou19815) [inline dti-website/src/pages/team.tsx:33]: solved?

File: dti-website/src/pages/team.tsx [hunk 3]
```diff
@@ -24,13 +24,13 @@ export default function TeamPage(): JSX.Element {
     role = '',
     isLead = false
   ): readonly { info: NovaMember; id: string }[] =>
-    members
+    (members as NovaMember[])
```
Comment (SamChou19815) [inline dti-website/src/pages/team.tsx:27]: why this cast?

File: dti-website/src/components/DtiProject.tsx [hunk 4]
```diff
@@ -20,11 +20,11 @@ export default function DtiProject({
...
   if (project == null) throw new Error();
-  const pastMembers = allMembers.filter((member) =>
-    (member.otherSubteams || []).includes(teamId)
+  const pastMembers = (allMembers as NovaMember[]).filter((member) =>
+    ((member.formerSubteams as string[]) || []).includes(teamId)
   );
-  const currentMembers = allMembers.filter(
-    (member) => member.subteam === teamId
+  const currentMembers = (allMembers as NovaMember[]).filter((member) =>
+    member.subteams ? (member.subteams as string[]).includes(teamId) : false
```
Comment (SamChou19815) [inline dti-website/src/components/DtiProject.tsx:27]: why do we need cast here

File: dti-website/src/pages/team.tsx (hunk 3 above)
Comment (henry-li-06) [reply dti-website/src/pages/team.tsx:27]: i think this and the other casts are unnecessary actually

Comment (henry-li-06) [reply dti-website/src/pages/team.tsx:27]: oh just kidding, without these casts i was getting some type errors but not sure if there is a better way to resolve that

File: dti-website/src/pages/team.tsx (hunk 2 above)
Comment (henry-li-06) [reply dti-website/src/pages/team.tsx:33]: I was getting `Unused '@ts-expect-error' directive.`. Should this just be removed?

File: dti-website/src/pages/team.tsx (hunk 3 above)
Comment (SamChou19815) [reply dti-website/src/pages/team.tsx:27]: I think it's caused by the `hello DTI` in the member. Deleting that and things should work without casts.

File: dti-website/src/components/DtiProject.tsx [hunk 5]
```diff
@@ -21,10 +21,10 @@ export default function DtiProject({
   const project = allProjects.find((it) => it.teamId === teamId);
   if (project == null) throw new Error();
   const pastMembers = allMembers.filter((member) =>
-    (member.otherSubteams || []).includes(teamId)
+    ((member.formerSubteams as string[]) || []).includes(teamId)
   );
-  const currentMembers = allMembers.filter(
-    (member) => member.subteam === teamId
+  const currentMembers = allMembers.filter((member) =>
+    member.subteams ? (member.subteams as string[]).includes(teamId) : false
```
Comment (SamChou19815) [inline dti-website/src/components/DtiProject.tsx:27]: probably don't need cast here

**Review by SamChou19815 — APPROVED** [review]
Final nit. Otherwise looks good.

File: dti-website/src/components/DtiProject.tsx (hunk 5 above)
Comment (henry-li-06) [reply dti-website/src/components/DtiProject.tsx]: Getting `Argument of type 'string' is not assignable to parameter of type 'never'` without the cast

**Review by SamChou19815 — APPROVED** [review]

**Pull request merged**